### DISCUSSION TITLE
Display name for tasks

### DISCUSSION THREAD

**Discussion by mgzenitech** [nat-n/poethepoet#141] [Ideas]
Hello!

So currently with `verbosity = 0` we have commands output for each task, however for production use there is no point in displaying everything, i.e. setting `verbosity = -1` works best. BUT then if there are sequences within sequences it becomes really hard to track what's going on. Maybe it would be a quick fix just to add optional `display_name` next to each task that would be shown with `verbosity = -1`? Or just output the task name?

For example if the task names would be output:

```toml
[tool.poe.tasks.lint]
sequence = [
  "lint:dockerfile",
  "lint:markdown",
  "lint:poetry",
  "lint:versions",
  "lint:yaml"
]
ignore_fail = "return_non_zero"

[tool.poe.tasks."lint:yaml"]
sequence = [
  "lint:yaml:a",
  "lint:yaml:b",
  "lint:yaml:c"
]
ignore_fail = "return_non_zero"
```

would output something like:
```
lint → lint:dockerfile:
...output from task...

lint → lint:markdown:
...output from task...

lint → lint:poetry:
...output from task...

lint → lint:versions:
...output from task...

lint → lint:yaml → lint:yaml:a:
...output from task...

lint → lint:yaml → lint:yaml:b:
...output from task...

lint → lint:yaml → lint:yaml:c:
...output from task...
```

**Comment by nat-n** [comment]
Hi @mgzenitech, I've been thinking I need to revisit how task output is presented, so this is useful feedback, thanks!

Thinking aloud here:

I agree that the current way that it outputs the just the index of the task in the sequence is not very useful, and outputting the whole task content maybe a bit too much sometimes. 

I like your idea of prefixing the subtask of the sequence with the sequence task nam e.

So currently quiet mode means no extra output, and I think that probably still makes sense, but maybe it should only output multiline tasks in verbose mode?

However including task content in the output for simple cases still seems useful to me, a bit like what make does...

**Further suggestions are welcome!**

**Comment by mgzenitech** [comment]
Probably it would be best to introduce additional verbosity level:

```
1 - max
0 - current default
-1 - only task names (as in my example, even non-sequence tasks would display the name)
-2 completely silent
```

OR move current default to `1` and `0` (default) would be just the task names

```
1 - current default
0 - only task names (as in my example, even non-sequence tasks would display the name)
-1 - completely silent
```

That way, everyone would benefit. What do you think?

**Comment by mgzenitech** [comment]
There's one more issue though:

```toml
[tool.poe.tasks.lint]
sequence = [
  { shell = "echo 'what to do here?'" },
  "lint:dockerfile",
  "lint:markdown",
  "lint:poetry",
  "lint:versions",
  "lint:yaml"
]
ignore_fail = "return_non_zero"

[tool.poe.tasks."lint:yaml"]
sequence = [
  "lint:yaml:a",
  "lint:yaml:b",
  "lint:yaml:c"
]
ignore_fail = "return_non_zero"
```

So my suggestion would probably be to also introduce the index, i.e.:
```
Executing: lint[0]
...output from task...

Executing: lint[1] → lint:dockerfile
...output from task...

Executing: lint[2] → lint:markdown
...output from task...

Executing: lint[3] → lint:poetry
...output from task...

Executing: lint[4] → lint:versions
...output from task...

Executing: lint[5] → lint:yaml[0] → lint:yaml:a
...output from task...

Executing: lint[5] → lint:yaml[1] → lint:yaml:b
...output from task...

Executing: lint[5] → lint:yaml[2] → lint:yaml:c
...output from task...
```

**Comment by mgzenitech** [comment]
hey @nat-n ! any updates on this? :)

**Comment by nat-n** [comment]
Hi @mgzenitech,

Thanks for the additional feedback. I'm going move this discussion to an issue to bump it up on my todo list but it's not on deck for me at the moment (and my capacity is quite limited lately). If you'd be interested in working on implementing some changes here then I'd be happy to discuss to refine the design.

I'm unsure how best to handle the verbosity issue. I understand the desire to have a "less verbose but still informative" output mode, though neither of the two obvious options is ideal. I think showing the full task content (maybe with better formatting) is the optimal default behaviour (for verbosity 0). However it could be a little bit of a disruptive change to the behaviour of the `-q` flag, by requiring `-qq` to actually get it to be quiet — though this is probably the better option.
